tests/qa_automation/execute_test_run.pm

```diff
@@ -21,17 +21,14 @@ use upload_system_log;
 use base "opensusebasetest";
 
 sub run {
-    my $timeout   = get_var("MAX_JOB_TIME", 7200);
-    my $test      = get_var("QA_TESTSUITE") . get_var("QA_VERSION");
-    my $runfile   = "/usr/share/qa/tools/test_$test-run";
-    my $run_exist = script_run("ls $runfile | wc -l");
-    unless ($run_exist eq "1") {
-        $runfile = "/usr/lib/ctcs2/tools/test_$test-run";
-    }
-    my $run_log = "/tmp/$test-run.log";
+    my $timeout  = get_var("MAX_JOB_TIME", 9600) - 1200;              #deduct 20 minutes for previous steps, due to poo#30183
```

Comment:
I wonder if we can end up with negative timeout here. I also believe that by default job will fail after 2 hours (7200 sec), so not sure if it will fix an issue mentioned in poo#30183

Reply:
The point in poo#30183 is not solve the fail in description, because if no issue happened test will finish in around 1 hour. Oliver care about the timeout which we set in the middle of the test should shorter than whole test timeout. I agree with him and it's reasonable. The difficult part is different test has different prepare time, this is a simple solution, not the best.

Reply:
You suggestion is also valuable, I'll add a condition to see if MAX_JOB_TIME bigger than 1200 or not, thank you :)